src/data/oppfolgingsplan/oppfolgingsplanForesporselHooks.ts

```diff
@@ -42,14 +46,18 @@ export function usePostOppfolgingsplanForesporsel() {
   const path = `${ISOPPFOLGINGSPLAN_ROOT}/oppfolgingsplan/foresporsler`;
   const postOppfolgingsplanForesporsel = (
     foresporsel: NewOppfolgingsplanForesporselDTO
-  ) => post<NewOppfolgingsplanForesporselDTO>(path, foresporsel);
+  ) => post<OppfolgingsplanForesporselResponse>(path, foresporsel);
 
   return useMutation({
     mutationFn: postOppfolgingsplanForesporsel,
-    onSuccess: () => {
-      return queryClient.invalidateQueries({
-        queryKey: oppfolgingsplanQueryKeys.foresporsel(personident),
-      });
+    onSuccess: (createdForesporsel: OppfolgingsplanForesporselResponse) => {
+      queryClient.setQueryData(
+        oppfolgingsplanForesporselQueryKeys.foresporsel(personident),
+        (oldData: OppfolgingsplanForesporselResponse[]) => [
+          createdForesporsel,
+          ...oldData,
+        ]
+      );
```

Comment:
For at dette skal fungere uten sjekker så tror jeg backenden må returnere 200 og tom liste. Av hva jeg ser av meninger og retninger på nettet ser også dette ut som den mest riktige responsen. Ser vi returnerer 204 i dag, så legger inn en PR på det om folk er enige 😊
Mange steder invaliderer vi bare cachen mens vi her setter verdien på nytt. Det går an å lese mer om dette [her](https://tkdodo.eu/blog/mastering-mutations-in-react-query#awaited-promises), som Anders sendte til meg.

Reply:
For at det skal funke må Request og Response DTO være helt like da, eller? Ref linje 49.

Reply:
Må ha vært en feil der før ja, ser riktig ut nå :+1:  Litt forvirrende at `post<T>` er responsen, men request og response skal i hvert fall ikke trenge å være like.